posts/2021/12/conda-recipe-grayskull.md

```diff
@@ -0,0 +1,118 @@
+During the summer of 2021 I was an intern at Quansight Labs and I worked on a project called ‘Grayskull’.
+Before we learn about Grayskull, as promised, I’ll talk about software packages and Conda.
+
+## Software Packages
+A software package is, simply put, a working piece of code that somebody wrote and published for others to use. A functional packet of code that does something — a package. You install packages for your use. Sometimes you import them into your code while writing a package of your own.
+
+```
+import numpy
+```
+That’s you ‘importing’ the package numpy into your code.
+
+## Channels
+Once you write a package you might want to publish it so that others can download it and use it. Packages are published on ‘channels’.
+Channels are like warehouses of packages.
+
+## Conda
+Conda is an OS-agnostic package manager with great popularity in the Python world and data science adjacent libraries.
+
+Conda-build is a set of commands and tools that lets you build your own packages for Conda. These tools let you manage the environments and dependencies of your packages and generate the needed context for your project.
+
+Anaconda provides a default channel called ‘defaults’ where packages are published. There are several community driven channels as well, conda-forge being the most popular one.
+From here onwards in this blog we’ll assume that we want to publish our package on the conda-forge channel.
+
+## Publishing a package on conda-forge
+Publishing packages on the conda-forge channel requires the knowledge of ‘recipes’. A recipe is a collection of files that defines how to build a package. Minimally a recipe contains a meta.yaml file that describes:
+- The package name and version
+- its dependencies
+- how to build it
+- some other metadata
+You can learn more about recipes [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#meta-yaml).
+
+<p align="center">
+    <img
+     alt="Cute representation of a package recipe"
+     src="/images/2021/12/recipe_animation.png">
+    <i><br>Cute representation of a package recipe</i>
+</p>
+
+To publish your package write its recipe and create a pull request on the [staged-recipes](https://github.com/conda-forge/staged-recipes) repository of conda-forge. This pull request will be community reviewed and if approved, your package will become available on the conda-forge channel.
+Under the hood, the recipe that you submit is fed to Conda-build which ultimately generates the package.
+
+<p align="center">
+    <img
+     alt="Conda-build transforms recipes into packages"
+     src="/images/2021/12/conda-build.png">
+    <i><br>Conda-build transforms recipes into packages</i>
+</p>
+
+## Grayskull — the automatic Conda recipe generator
+The process to publish a package on conda-forge is simple and straight forward: write the recipe, create a pull request, wait for the community review. But writing recipes is not simple. It can be error prone and tiresome.
```

Comment:
```suggestion
We have seen that the process to publish a package on conda-forge is simple and straight forward: write the recipe, create a pull request, wait for the community review. But writing recipes is not simple. It can be error prone and tiresome.
```